microsoft/testsuites/core/storage.py

```diff
@@ -345,6 +345,42 @@ def hot_add_disk_parallel_premium_ssd(self, log: Logger, node: Node) -> None:
             log, node, DiskType.PremiumSSDLRS, self.DEFAULT_DISK_SIZE_IN_GB
         )
 
+    @TestCaseMetadata(
+        description="""
+        This test case will verify mount azure nfs 4.1 on guest successfully.
+        """,
+        timeout=TIME_OUT,
+        requirement=simple_requirement(supported_features=[Nfs]),
+        priority=2,
+    )
+    def verify_azure_file_share_nfs(self, log: Logger, node: Node) -> None:
+        nfs = node.features[Nfs]
+        server_shared_dir = nfs.create()
+        storage_account_name = server_shared_dir.split("/")[0]
+        mount_nfs = f"{storage_account_name}.file.core.windows.net"
+        mount_dir = f"/mount/{server_shared_dir}"
+        max_retry = 10
+        retry = 0
+        mount_success = False
+        # retry if the dns name can't be resolved after creating shortly
+        while retry < max_retry:
```

Comment:
Use timeout, instead of counting, it's easier to measure by time.

Reply:
updated to use timeout